nimsuggest/src/parsing.nim

```diff
@@ -0,0 +1,59 @@
+## Handles parsing of the input to nimsuggest
+import strutils, parseutils
+
+import compiler/[options, pathUtils]
+
+import types
+
+const seps* = {':', ';', ' ', '\t'}
+
+proc parseQuoted(cmd: string; outp: var string; start: int): int =
+  var i = start
+  i += skipWhitespace(cmd, i)
+  if i < cmd.len and cmd[i] == '"':
+    i += parseUntil(cmd, outp, '"', i+1)+2
+  else:
+    i += parseUntil(cmd, outp, seps, i)
+  result = i
+
+proc parseCommandLine*(cmdLine: string; projectFull: string): CommandData =
+  ## Parses an input line to nimsuggest and returns a CommandData object with the content
+
+  var cmdDat = CommandData()
+
+  #We parse the input line in chunks using
+  #parseIdent to move us forward by one token at a time
+  var opc = ""
+  var i = parseIdent(cmdLine, opc, 0)
+# TODO: Repl this with a call to parseCommand
+  let cmdString = opc
+
+  let ideCmd = parseIdeCmd cmdString
+  cmdDat.ideCmd = ideCmd
+  #If the command cannot be parsed return it to the caller
+  if ideCmd == ideNone:
+    cmdDat.ideCmdString = cmdString
+    return cmdDat
```

Comment:
I call the parseIdeCmd function from the parser code instead of re-implementing it. However nimsuggest still has some extra commands that won't be parsed by the compilers parser. In that case we return those early, so they can be dealt with.